File: CodeForces/1462c/main.py
def inp():
    return(int(input()))
def inlt():
    return(list(map(int,input().split())))


def solve() :

    n = inp()

    if(n > 45) :
        print("-1")
        return
    
    if(n >= 1 and n <= 9) :
        print(n)
        return
    
    used = [False] * 9
    ans = []

    for i in range(9, 0, -1) :
        temp = n - i
        if(not used[i-1] and temp >= 0) :
            n -= i
            ans.append(str(i))
            used[i-1] = True
   
    ans.sort()
    ans = int(''.join(ans))
    print(ans)
    return

t = inp()

for _ in range(t) :
    solve()

File: CodeForces/1462c/main.cpp
#include <iostream>
#include <vector>

using namespace std;

typedef long long ll;
typedef long double ld;


void solve() {
    
    int n;
    cin >> n;

    if(n > 45) {
        cout << "-1\n";
        return;
    }
    
    if(n >= 1 && n <= 9) {
        cout << n << "\n";
        return;
    }
    
    bool used[9] = {false};
    vector<int> ans;

    for(int i = 9; i >= 1; i--) {
        int temp = n - i;
        if(!used[i-1] && temp >= 0) {
            n -= i;
            ans.emplace_back(i);
            used[i-1] = true;
        }
    }

    for(int i = ans.size() - 1; i >= 0; i--) {
        cout << ans[i] << "";
    }

    cout << "\n";
    return;
}

int main() 
{

    int t;
    cin >> t;

    while(t--) {
        solve();
    }
    
    return 0;
}
